core/src/main/scala/com/rallyhealth/vapors/factfilter/data/Fact.scala

```diff
@@ -54,15 +65,28 @@ object Fact {
   }
 }
 
-final case class TypedFact[A](
-  typeInfo: FactType[A],
-  value: A,
-) extends Fact {
+sealed trait TypedFact[A] extends Fact {
   type Value = A
 }
 
 object TypedFact {
 
+  final def apply[A](
+    typeInfo: FactType[A],
+    value: A,
+  ): TypedFact[A] = SourceFactOfType(typeInfo, value)
+
+  final def apply[A](
+    typeInfo: FactType[A],
+    value: A,
+    evidence: Evidence,
+  ): TypedFact[A] with DerivedFact = DerivedFactOfType(typeInfo, value, evidence)
+
+  final def unapply[A](fact: TypedFact[A]): Some[(FactType[A], A)] = fact match {
+    case SourceFactOfType(typeInfo, value) => Some((typeInfo, value))
+    case DerivedFactOfType(typeInfo, value, _) => Some((typeInfo, value))
```

Comment:
Does this lose important information?

Reply:
If it were important, the caller would use the `DerivedFact` extractor object.